src/Microsoft.AspNetCore.Http.Abstractions/Extensions/ResponseTrailerExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Http
+{
+    public static class ResponseTrailerExtensions
+    {
+        private const string Trailer = "Trailer";
+
+        public static void DeclareTrailer(this HttpResponse response, string trailerName)
+        {
+            response.Headers.AppendCommaSeparatedValues(Trailer, trailerName);
+        }
+
+        public static void AppendTrailer(this HttpResponse response, string trailerName, StringValues trailerValues)
+        {
+            var feature = response.HttpContext.Features.Get<IHttpResponseTrailersFeature>();
+            feature?.Trailers?.Append(trailerName, trailerValues);
```

Comment:
Checking for null when you're already checking for the feature existence seems like it might be a little too defensive. Technically response.Headers could be set to null too, but there's no check in DeclareTrailer.

Is the idea that the server might set the feature but leave the IHeaderDictionary null? Or are you just defending against some middleware setting Trailers to null?

Reply:
Just paranoia. We make a lot of assumptions about the request and response features because nothing would work without them, but for these very optional features I try not to assume anything.

Reply:
Is it weird that this can noop? Should it throw if trailers aren't supported?

Reply:
If it threw you'd need some other way to check for support before calling it. Checking for the presence of the feature is the simplest, but negates using an extension like this. I could add another extension here SupportsTrailers to check for the feature.

We also discussed some Http/1.1 scenarios where the feature may be present but the headers may be discarded with a log message (e.g. content-length)

Reply:
Err hmm, I'm not sure if I like that, it's very similar to adding headers after the response body has started. Today we throw. I like that model.